docs/source/branch.rst

```diff
@@ -8,125 +8,167 @@ The Branches File
-mapping that dictates the number of draws to run.
+To do this, we can use the ``input_draw_count`` key in a :term:`branch configuration<Branch Configuration>`.
+This key refers to an integer that represents the number of different input draws to generate simulations from.
 
 .. code-block:: yaml
+    :caption: parameter_uncertainty_branches.yaml
 
     input_draw_count: 10
 
+When we use this branch configuration along with the original :term:`model specification<Model Specification>`,
+we'll launch 10 simulations in parallel, each using a different set of input paramemters represented by the
+draw number.
+
+.. code-block:: sh
+
+  psimulate run /path/to/model_specification.yaml /path/to/parameter_uncertainty_branches.yaml
+
+
+.. note::
+
+  ``psimulate`` randomly selects the input draws it uses from the range [0, 999].  The selection
+  happens without replacement, so specifying an ``input_draw_count`` of 10 guarantees you
+  10 unique input draws.
+
+
 Stochastic Uncertainty
 ^^^^^^^^^^^^^^^^^^^^^^
-Vivarium simulations are probabilistic in nature. They tend to contain distributions that describe quantities rather
-than singular values. Because of this, our models are subject to stochastic uncertainty. To attempt to capture some of
-this uncertainty we can run the simulation with different random seeds which will result in different numbers drawn from
-the underlying pseudo-random number generator, and thus different output results. This can be specified in the branches
-file with the ``random_seed_count`` key. This key specifies an integer that represents the number of different random
-seeds to use, each generated randomly and run in a separate simulation.
+Vivarium simulations are probabilistic in nature. They use Monte Carlo sampling techniques to make decisions about
+who gets sick, who goes to the hospital, who dies, etc. This usage of randomness means our models have to
+consider the impact of :term:`stochastic uncertainty<Stochastic Uncertainty>` on its outputs.
 
-.. note::
-    Random seeds are a convenient way to scale up a simulation's population in parallel. For example, running a
-    simulation with one million simulants and a single random seed is equivalent to running the same simulation with
-    ten thousand people and 100 random seeds. However, because simulations specified with different seeds will be run
-    in parallel (link psimulate), this is often preferable.
+There are two ways to handle stochastic uncertainty. The first is to increase the size of the population you're
+simulating. This will wash out outlier cases that might heavily skew your results. This works fine up to a point,
+but simulation run time scales directly with the size of the population you're simulating.  Alternatively,
+you can run multiple simulations with different :term:`random seeds<Random Seed>` and aggregate your results across
+those simulations. This second approach takes advantage of parallel computing to keep run times under control.
```

Comment:
I think this could be spelled out a little more explicitly. It doesn't clearly say that each of the multiple sims can be smaller and thus run faster but that the final aggregation is the same as the big sim. Maybe add back the example comparing a sim of 1m people and 1 seed versus 10k people + 100 random seeds - I'm not sure that this paragraph totally makes it clear that running multiple sims with diff seeds allows essentially splitting your pop.